RoddyCore/src/de/dkfz/roddy/config/ToolFileGroupParameter.groovy

```diff
@@ -40,11 +53,38 @@ class ToolFileGroupParameter extends ToolEntry.ToolParameter<ToolFileGroupParame
         this.indexOptions = indexOptions
     }
 
-
+    /**
+     * Copy constructor
+     */
+    @Deprecated
+    ToolFileGroupParameter(String scriptParameterName, Class<FileGroup> groupClass, Class<BaseFile> genericFileClass, PassOptions passOptions, IndexOptions indexOptions, List<ToolFileParameter> files) {
+        super(scriptParameterName)
+        this.groupClass = groupClass
+        this.genericFileClass = genericFileClass
+        this.passOptions = passOptions
+        this.indexOptions = indexOptions
+        this.files = files
+    }
 
     @Override
     ToolFileGroupParameter clone() {
-        return new ToolFileGroupParameter(groupClass, genericFileClass, scriptParameterName, passOptions)
+        return new ToolFileGroupParameter(scriptParameterName, groupClass, genericFileClass,  passOptions, indexOptions, files?.collect { ToolFileParameter tfp -> tfp.clone() })
+    }
+
+    @Override
+    boolean equals(o) {
```

Comment:
Note that ToolParameter does not even have an .equals() method defined by you. Therefore it will use the reference and not the scriptParameterName to determine equality.